src/Mod/Draft/draftutils/init_draft_statusbar.py

```diff
@@ -164,21 +166,131 @@ def init_draft_statusbar(sb):
     """
     this function initializes draft statusbar
     """
+    param = App.ParamGet("User parameter:BaseApp/Preferences/Mod/Draft")
+
+    # SNAP WIDGET - init ----------------------------------------------------
+    
+    snap_widget = QtGui.QToolBar()
+    snap_widget.setObjectName("draft_snap_widget")
+    snap_widget.setIconSize(QtCore.QSize(16,16))
+    
+    # GRID BUTTON - init
+    gridbutton = QtGui.QPushButton(snap_widget)
+    gridbutton.setIcon(QtGui.QIcon.fromTheme("Draft",
+                                             QtGui.QIcon(":/icons/"
+                                                         "Draft_Grid.svg")))
+    gridbutton.setToolTip(QT_TRANSLATE_NOOP("Draft",
+                                            "Toggles Grid On/Off"))
+    gridbutton.setObjectName("Grid_Statusbutton")
+    gridbutton.setWhatsThis("Draft_ToggleGrid")
+    gridbutton.setFlat(True)
+    QtCore.QObject.connect(gridbutton,QtCore.SIGNAL("clicked()"),
+                           lambda f=Gui.doCommand, 
+                           arg='Gui.runCommand("Draft_ToggleGrid")':f(arg))
```

Comment:
I'm not sure of the syntax here. Is it supposed to be
`Gui.doCommand('Gui.runCommand("Draft_ToggleGrid")')`

This looks convoluted. I think you mean to run just `Gui.runCommand("Draft_ToggleGrid")`.

Reply:
well, the fact is that I wanted the string `Gui.runCommand("Draft_ToggleGrid")`  to be printed into the console... like if you were activating the command straight from the UI.